### PR TITLE
chore: Configure GitHub Actions for CI

### PR DESCRIPTION
**Pull request by bf4**
based on .travis.yml

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: [ main ]
```
Comment (bf4) [inline .github/workflows/ci.yaml:6]: shrug

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.1
+          bundler-cache: true
+      # - name: Run RuboCop
+      #   run: bundle exec rubocop
```
Comment (bf4) [inline .github/workflows/ci.yaml:19]: for laters?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  rspec:
+    runs-on: ubuntu-latest
+    env:
+      RUBYOPT: "-W:deprecated" # show warnings, deprecations only
```
Comment (bf4) [inline .github/workflows/ci.yaml:24]: it can't hurt

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      FARADAY_VERSION: ${{ matrix.faraday }}
```
Comment (bf4) [inline .github/workflows/ci.yaml:25]: if wanted

File: .github/workflows/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Set update schedule for GitHub Actions
+
+version: 2
+updates:
+
+  - package-ecosystem: "github-actions"
```
Comment (bf4) [inline .github/workflows/dependabot.yml:6]: if wanted

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (minimul) [reply .github/workflows/ci.yaml:19]: Nah.

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (minimul) [reply .github/workflows/ci.yaml:24]: 👍🏻

File: .github/workflows/dependabot.yml (hunk 5 above)
Comment (minimul) [reply .github/workflows/dependabot.yml:6]: 👍🏻

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (minimul) [reply .github/workflows/ci.yaml:25]: 👍🏻

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (minimul) [reply .github/workflows/ci.yaml:6]: Well I'm **not** going to rename to `main`.

File: .github/workflows/dependabot.yml (hunk 5 above)
Comment (minimul) [reply .github/workflows/dependabot.yml:6]: Thanks @bf4

**Comment by bf4** [comment]
@minimul updated per review and force pushed

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ '2.7', '3.0', '3.1', '3.2' ]
+        faraday: [ '~> 1.0', '~> 2.0' ]
```
[the remark this answers is not in the document]
Comment (minimul) [reply .github/workflows/ci.yaml:18]: Yes, let's support `Faraday 1.x` in the CI.

We'll soon go to `>= 2.0` but this is good in the meantime.

[the remark this answers is not in the document]
Comment (bf4) [reply .github/workflows/ci.yaml:18]: Any other changes to get this merged?

[the remark this answers is not in the document]
Comment (minimul) [reply .github/workflows/ci.yaml:18]: Nope. Merged. Thanks again.

**Pull request merged**